mermaid_py/client.py

```diff
@@ -0,0 +1,443 @@
+        # TODO: Handle pagination
+        if info == "me" and not self.authenticated:
+            raise Exception('Authentication Required for access to "me" endpoint')
+
+        if info in self.npr_endpoints:
+            return self._fetch_resource(info)
+        else:
+            raise Exception(f"get_info() invalid info parameter:{info}")
+
+    def get_choices(self):
+        """
+        Gets a list of objects, each one of which has a name item (e.g., countries) and a data item that is a list
+        of available choice objects.
+        :return: list of choice objects.
+        :rtype: list
+        """
+        # TODO: Filter: /updates/?timestamp="DATE"
+        return self._fetch_resource("choices")
+
+    def get_attribute(self, attr):
+        """
+        Gets attributes which are “things that can be observed” – coral and other taxa as well as non-organic
+        benthic substrates for benthic transects and bleaching surveys, fish species/genera/families as well
+        as arbitrary fish groupings for fish belt transects, and so on.
+        :param attr: fishsizes, benthicattributes, fishfamilies, fishgenera, fishspecies, fishgroupings.
+        :type attr: str
+        :return: attribute data.
+        :rtype: dict
+        """
+        if attr not in self.attrs_endpoints:
+            raise Exception(f"Invalid attribute: {attr}")
+        return self._fetch_resource(attr)
+
+    def get_projects(self, showall=False):
+        """
+        Gets the projects subdirectory. Without query parameters, returns a list of projects of which
+        the user is a member. The showall query parameter may be used to return projects unfiltered by the user’s
+        membership. showall is important when the user is unauthenticated.
+        :param showall: (optional) Returns all projects unfiltered by the user’s membership.
+        :type showall: bool
+        :return: projects.
+        :rtype: dict
+        """
+        if not showall:
+            return self._fetch_resource("projects")
+        else:
+            payload = "showall"
+            return self._fetch_resource("projects", parameters=payload)
+
+    def get_my_project(self, name=None, id=None):
+        """
+        Gets a specific MERMAID project. Either name or id required.
+        See https://mermaid-api.readthedocs.io/en/latest/projects.html for more information on project resources.
+        :param name: (optional if 'id' provided) MERMAID project name.
+        :type name: str
+        :param id: (optional if 'name' provided) MERMAID project ID.
+        :type id: str
+        :return: project with associated data.
+        :rtype: dict
+        """
+        if id:
+            return self._api_projects_path(id=id)
+        elif name:
+            projects = self.get_projects(showall=True)
+            proj = get_dict_by_keyval("name", name, projects["results"])
+            if proj is None:
+                raise Exception(f"No project with the name:{name}")
```

Comment:
I think your question is correct, and the exception (here and elsewhere) should handle id in addition to name (since you're handling both in all other contexts).